### ISSUE TITLE
Language server doesn't stop when quitting VSCode (Linux)

### ISSUE THREAD

**Issue by sporto** [elm-tooling/elm-language-client-vscode#203]
I noted that after quitting VSCode the language server keeps running on the background.

## Expected Behavior
Stop the server process

## Current Behavior
![image](https://user-images.githubusercontent.com/1005498/127079941-31d866e1-6479-47e1-8271-5c929bda9a1a.png)

I open VS code, the LS starts. Then I quit the LS keeps running. Doesn't happen all the time, but often enough.
I have to kill the process manually.

## Steps to Reproduce (for bugs)
- Using Ubuntu 21.04
- Open VS code, edit Elm code for a while
- Quit VS code

## Your Environment

* Version used: 2.2.0
* Editor name and version (e.g. VSCode 1.36.1):  VSCode 1.58.2
* Environment name and version (e.g. node.js 5.4): node 14.15.1
* Operating System and version: PopOS 21.04 (based on Ubuntu 21.04)

Thanks

**Comment by razzeee** [comment]
Is the editing still working for you at that point in time?

**Comment by sporto** [comment]
@razzeee I don't understand the question, sorry. Maybe I wasn't clear. After quitting VS Code the elm language server still keeps a process running. I would expect that quitting VS code would kill the lsp process. Happy to provide more info and help.

**Comment by razzeee** [comment]
It would be interesting, if you could still use the lsp before that. If it was already stuck on something at that point, it might explain, why it misses the shutdown.

**Comment by sporto** [comment]
I can confirm that this happens when the LSP stops responding.

I don't know how it gets into this state, it appears that this happens when I change things too quickly, but unsure. I notice this when the LSP stops formatting my code on save.

Then if I quit the node process keeps going and I need to kill it manually. So the cause of the issue that the LSP becomes unresponsive.

**Comment by razzeee** [comment]
Yeah, that's going to be harder to track down, but good, that we can narrow it down.
